Dockerfile.nccl

```diff
@@ -0,0 +1,87 @@
+# Start with the NVIDIA CUDA base image
+FROM nvidia/cuda:12.2.0-devel-ubuntu22.04@sha256:7e81c70fb55bfc54e6711700eb721bb94159a3cd546620b37210d12c17914fa6
+
+ARG EFA_INSTALLER_VERSION=latest
+ARG AWS_OFI_NCCL_VERSION=1.7.2
+ARG NCCL_TESTS_VERSION=master
+
+# Install necessary dependencies
+RUN apt-get update -y
+RUN apt-get remove -y --allow-change-held-packages \
+                      libmlx5-1 ibverbs-utils libibverbs-dev libibverbs1 \
+                      libnccl2 libnccl-dev
+
+RUN rm -rf /opt/hpcx \
+    && rm -rf /usr/local/mpi \
+    && rm -rf /usr/local/ucx \
+    && rm -f /etc/ld.so.conf.d/hpcx.conf \
+    && ldconfig
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated \
+    sudo git \
+    gcc \
+    vim \
+    kmod \
+    openssh-client \
+    openssh-server \
+    build-essential \
+    wget curl \
+    autoconf \
+    libtool \
+    gdb \
+    automake \
+    python3-distutils \
+    cmake \
+    apt-utils \
+    devscripts \
+    debhelper \
+    libsubunit-dev \
+    check \
+    pkg-config \
+    libhwloc-dev
+
+RUN mkdir -p /var/run/sshd
+RUN sed -i 's/[ #]\(.*StrictHostKeyChecking \).*/ \1no/g' /etc/ssh/ssh_config && \
+    echo "    UserKnownHostsFile /dev/null" >> /etc/ssh/ssh_config && \
+    sed -i 's/#\(StrictModes \).*/\1no/g' /etc/ssh/sshd_config
+ENV LD_LIBRARY_PATH /opt/amazon/openmpi/lib64:/opt/amazon/openmpi/lib:/opt/amazon/efa/lib64:/opt/aws-ofi-nccl/install/lib:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/lib/:/usr/lib64:/usr/lib/x86_64-linux-gnu/:$LD_LIBRARY_PATH
+ENV PATH /usr/local/cuda/bin:/opt/amazon/openmpi/bin:/opt/amazon/efa/bin:/usr/sbin:/usr/bin:/usr/local/bin:$PATH
+
+# Install EFA
+RUN cd $HOME \
+    && curl -O https://efa-installer.amazonaws.com/aws-efa-installer-${EFA_INSTALLER_VERSION}.tar.gz \
+    && tar -xf $HOME/aws-efa-installer-${EFA_INSTALLER_VERSION}.tar.gz \
+    && cd aws-efa-installer \
+    && ./efa_installer.sh -y -g -d --skip-kmod --skip-limit-conf --no-verify \
```

Comment:
`--no-verify` should be removed, see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start.html#efa-start-enable

Reply:
@Issacwww what does that flag do? sounds like it may require an EFA device to be attached or something. We're also skipping the kmod installation, since all we need in the container image is the userspace libraries

Reply:
The doc use the command `sudo ./efa_installer.sh -y` to install efa. Should `-g -d --skip-kmod --skip-limit-conf --no-verify` also be removed?

Reply:
@cartermckinnon detail about the flags
```
 -d, --debug-packages   Install debug packages
 -u, --uninstall        Uninstall EFA packages
 -y, --yes              Run installer without prompting for confirmation
 -n, --no-verify        Skip EFA device verification and test
 -l, --skip-limit-conf  Skip EFA limit configuration
 -k, --skip-kmod        Skip EFA kmod installation
 -g, --enable-gdr       Enable GPUDirect RDMA support. This option is not needed as the support is enabled by default.
 -m, --minimal          Only install kernel module and rdma-core, do not install libfabric and mpi
 --mpi                  A comma-delimited list of MPI flavors(without space) to install. Valid choices are: openmpi4, openmpi5. Only effective if --minimal is not provided.
```

Reply:
>  -n, --no-verify        Skip EFA device verification and test

So, we definitely want to use that flag, because we won't have a real EFA device attached when we're just building the container image. I think all the other flags also make sense. All we need the EFA installer to do is put some libraries on the path in the container image